nginx/nginx.go

```diff
@@ -465,11 +465,17 @@ type ingressKey struct {
 	Host, Path string
 }
 
-func uniqueIngressEntries(ingressEntries controller.IngressEntries) []controller.IngressEntry {
-	sort.Sort(ingressEntries)
+func uniqueIngressEntries(entries controller.IngressEntries) []controller.IngressEntry {
+	sort.Slice(entries, func(i, j int) bool {
+		iEntry := entries[i]
+		jEntry := entries[j]
+		iString := strings.Join([]string{iEntry.Namespace, iEntry.Name, iEntry.Host, iEntry.Path}, "")
+		jString := strings.Join([]string{jEntry.Namespace, jEntry.Name, jEntry.Host, jEntry.Path}, "")
```

Comment:
You still need a separator here, something that won't be used in any of the strings. Otherwise "foo/bar" and "foob/ar" are equivalent.